libs/core/src/main/java/org/elasticsearch/core/ReleasableIterator.java

```diff
@@ -46,4 +46,29 @@ public String toString() {
 
         };
     }
+
+    /**
+     * Returns an empty iterator over the supplied value.
+     */
+    static <T extends Releasable> ReleasableIterator<T> empty() {
+        return new ReleasableIterator<>() {
+            @Override
+            public boolean hasNext() {
+                return false;
+            }
+
+            @Override
+            public T next() {
+                return null;
```

Comment:
We could assert false here? next should never be called in this case

Reply:
Sounds good!